src/services/transactions/hooks/useTrackTransactionStatus.ts

```diff
@@ -19,12 +25,27 @@ export interface UseTrackTransactionStatusArgsType {
 }
 
 export function useTrackTransactionStatus({
-  transactionId,
+  transactionId: txId,
   onSuccess,
   onFail,
   onCancelled,
   onTimedOut
 }: UseTrackTransactionStatusArgsType) {
+  const { signedTransactionsArray } = useGetSignedTransactions();
+  const isWalletProvider = getIsProviderEqualTo(LoginMethodsEnum.wallet);
+
+  const [sessionId] =
+    signedTransactionsArray.length > 0
+      ? signedTransactionsArray[signedTransactionsArray.length - 1]
+      : [];
+
+  /**
+   * Web wallet restores sessionId
```

Comment:
Can we expand the comment a bit, please? Maybe something like "when using web wallet, txId will be lost when redirected back, so we extract it from signedTransactions"